[2025-10-09T01:53:58.210+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-09T01:53:58.297+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream scheduled__2025-10-09T00:00:00+00:00 [queued]>
[2025-10-09T01:53:58.321+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream scheduled__2025-10-09T00:00:00+00:00 [queued]>
[2025-10-09T01:53:58.323+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2025-10-09T01:53:58.410+0000] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): run_spark_stream> on 2025-10-09 00:00:00+00:00
[2025-10-09T01:53:58.457+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62: DeprecationWarning: This process (pid=69) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-09T01:53:58.465+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'restaurant_analytics', 'run_spark_stream', 'scheduled__2025-10-09T00:00:00+00:00', '--job-id', '234', '--raw', '--subdir', 'DAGS_FOLDER/restaurant_dag.py', '--cfg-path', '/tmp/tmpcq40xtl4']
[2025-10-09T01:53:58.478+0000] {standard_task_runner.py:91} INFO - Job 234: Subtask run_spark_stream
[2025-10-09T01:53:58.486+0000] {standard_task_runner.py:64} INFO - Started process 81 to run task
[2025-10-09T01:53:58.849+0000] {task_command.py:426} INFO - Running <TaskInstance: restaurant_analytics.run_spark_stream scheduled__2025-10-09T00:00:00+00:00 [running]> on host 01b28d042a8a
[2025-10-09T01:53:59.234+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='quantruong1918@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='restaurant_analytics' AIRFLOW_CTX_TASK_ID='run_spark_stream' AIRFLOW_CTX_EXECUTION_DATE='2025-10-09T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-09T00:00:00+00:00'
[2025-10-09T01:53:59.251+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-09T01:53:59.261+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-10-09T01:53:59.268+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'export PYSPARK_PYTHON=/usr/local/bin/python && export PYSPARK_DRIVER_PYTHON=/usr/local/bin/python && /opt/spark/bin/spark-submit --master local[*] --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.5.1 /opt/***/dags/spark_streaming.py']
[2025-10-09T01:53:59.326+0000] {subprocess.py:86} INFO - Output:
[2025-10-09T01:54:05.760+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-10-09T01:54:06.167+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-10-09T01:54:06.171+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-10-09T01:54:06.206+0000] {subprocess.py:93} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
[2025-10-09T01:54:06.208+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-254fd245-f202-49cb-a971-a8430f3bfe9d;1.0
[2025-10-09T01:54:06.209+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-10-09T01:54:11.274+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 5070ms :: artifacts dl 0ms
[2025-10-09T01:54:11.275+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-10-09T01:54:11.276+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-10-09T01:54:11.277+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-10-09T01:54:11.278+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-10-09T01:54:11.279+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-10-09T01:54:11.280+0000] {subprocess.py:93} INFO - 	|      default     |   1   |   0   |   0   |   0   ||   0   |   0   |
[2025-10-09T01:54:11.281+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-10-09T01:54:11.282+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.283+0000] {subprocess.py:93} INFO - :: problems summary ::
[2025-10-09T01:54:11.284+0000] {subprocess.py:93} INFO - :::: WARNINGS
[2025-10-09T01:54:11.285+0000] {subprocess.py:93} INFO - 	problem while downloading module descriptor: https://repo1.maven.org/maven2/org/apache/spark/spark-parent_2.12/3.5.1/spark-parent_2.12-3.5.1.pom: impossible to move part file to definitive one: /home/***/.ivy2/cache/org.apache.spark/spark-parent_2.12/ivy-3.5.1.xml.original.part -> /home/***/.ivy2/cache/org.apache.spark/spark-parent_2.12/ivy-3.5.1.xml.original (126ms)
[2025-10-09T01:54:11.286+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.287+0000] {subprocess.py:93} INFO - 	io problem while parsing ivy file: https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_2.12/3.5.1/spark-sql-kafka-0-10_2.12-3.5.1.pom (java.io.IOException: Impossible to load parent for file:/home/***/.ivy2/cache/org.apache.spark/spark-sql-kafka-0-10_2.12/ivy-3.5.1.xml.original. Parent=org.apache.spark#spark-parent_2.12;3.5.1)
[2025-10-09T01:54:11.288+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.289+0000] {subprocess.py:93} INFO - 		module not found: org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.1
[2025-10-09T01:54:11.291+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.292+0000] {subprocess.py:93} INFO - 	==== local-m2-cache: tried
[2025-10-09T01:54:11.293+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.294+0000] {subprocess.py:93} INFO - 	  file:/home/***/.m2/repository/org/apache/spark/spark-parent_2.12/3.5.1/spark-parent_2.12-3.5.1.pom
[2025-10-09T01:54:11.296+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.297+0000] {subprocess.py:93} INFO - 	  -- artifact org.apache.spark#spark-parent_2.12;3.5.1!spark-parent_2.12.jar:
[2025-10-09T01:54:11.297+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.299+0000] {subprocess.py:93} INFO - 	  file:/home/***/.m2/repository/org/apache/spark/spark-parent_2.12/3.5.1/spark-parent_2.12-3.5.1.jar
[2025-10-09T01:54:11.299+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.300+0000] {subprocess.py:93} INFO - 	==== local-ivy-cache: tried
[2025-10-09T01:54:11.302+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.303+0000] {subprocess.py:93} INFO - 	  /home/***/.ivy2/local/org.apache.spark/spark-parent_2.12/3.5.1/ivys/ivy.xml
[2025-10-09T01:54:11.305+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.307+0000] {subprocess.py:93} INFO - 	  -- artifact org.apache.spark#spark-parent_2.12;3.5.1!spark-parent_2.12.jar:
[2025-10-09T01:54:11.308+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.312+0000] {subprocess.py:93} INFO - 	  /home/***/.ivy2/local/org.apache.spark/spark-parent_2.12/3.5.1/jars/spark-parent_2.12.jar
[2025-10-09T01:54:11.314+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.315+0000] {subprocess.py:93} INFO - 	==== central: tried
[2025-10-09T01:54:11.316+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.317+0000] {subprocess.py:93} INFO - 	  https://repo1.maven.org/maven2/org/apache/spark/spark-parent_2.12/3.5.1/spark-parent_2.12-3.5.1.pom
[2025-10-09T01:54:11.318+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.319+0000] {subprocess.py:93} INFO - 	==== spark-packages: tried
[2025-10-09T01:54:11.321+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.322+0000] {subprocess.py:93} INFO - 	  https://repos.spark-packages.org/org/apache/spark/spark-sql-kafka-0-10_2.12/3.5.1/spark-sql-kafka-0-10_2.12-3.5.1.pom
[2025-10-09T01:54:11.323+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.324+0000] {subprocess.py:93} INFO - 	  -- artifact org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.1!spark-sql-kafka-0-10_2.12.jar:
[2025-10-09T01:54:11.325+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.326+0000] {subprocess.py:93} INFO - 	  https://repos.spark-packages.org/org/apache/spark/spark-sql-kafka-0-10_2.12/3.5.1/spark-sql-kafka-0-10_2.12-3.5.1.jar
[2025-10-09T01:54:11.327+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.328+0000] {subprocess.py:93} INFO - 		::::::::::::::::::::::::::::::::::::::::::::::
[2025-10-09T01:54:11.329+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.330+0000] {subprocess.py:93} INFO - 		::          UNRESOLVED DEPENDENCIES         ::
[2025-10-09T01:54:11.331+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.332+0000] {subprocess.py:93} INFO - 		::::::::::::::::::::::::::::::::::::::::::::::
[2025-10-09T01:54:11.332+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.334+0000] {subprocess.py:93} INFO - 		:: org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.1: not found
[2025-10-09T01:54:11.336+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.338+0000] {subprocess.py:93} INFO - 		::::::::::::::::::::::::::::::::::::::::::::::
[2025-10-09T01:54:11.339+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.340+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.341+0000] {subprocess.py:93} INFO - 
[2025-10-09T01:54:11.342+0000] {subprocess.py:93} INFO - :: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
[2025-10-09T01:54:11.344+0000] {subprocess.py:93} INFO - Exception in thread "main" java.lang.RuntimeException: [unresolved dependency: org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.1: not found]
[2025-10-09T01:54:11.344+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmitUtils$.resolveMavenCoordinates(SparkSubmit.scala:1608)
[2025-10-09T01:54:11.345+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.DependencyUtils$.resolveMavenDependencies(DependencyUtils.scala:185)
[2025-10-09T01:54:11.346+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.prepareSubmitEnvironment(SparkSubmit.scala:334)
[2025-10-09T01:54:11.347+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:964)
[2025-10-09T01:54:11.348+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:194)
[2025-10-09T01:54:11.349+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:217)
[2025-10-09T01:54:11.350+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
[2025-10-09T01:54:11.351+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1120)
[2025-10-09T01:54:11.352+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1129)
[2025-10-09T01:54:11.353+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-10-09T01:54:11.354+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-10-09T01:54:11.356+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-09T01:54:11.378+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-10-09T01:54:11.388+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=restaurant_analytics, task_id=run_spark_stream, run_id=scheduled__2025-10-09T00:00:00+00:00, execution_date=20251009T000000, start_date=20251009T015358, end_date=20251009T015411
[2025-10-09T01:54:11.415+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 234 for task run_spark_stream (Bash command failed. The command returned a non-zero exit code 1.; 81)
[2025-10-09T01:54:11.453+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-10-09T01:54:11.494+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-09T01:54:11.500+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
